inductiva/scenarios.py

```diff
@@ -13,39 +12,56 @@
 class Scenario(ABC):
     """Base class for scenarios."""
 
+    def _setup_output_dir(self, output_dir: str):
+        """Setup the scenario output directory."""
+        if output_dir is None:
+            scenario_name_splitted = split_camel_case(self.__class__.__name__)
+            output_dir_prefix = "-".join(scenario_name_splitted).lower()
+            output_dir = get_timestamped_path(f"{output_dir_prefix}-output")
+        output_dir = resolve_path(output_dir)
+        return output_dir
+
+    def _setup_config(self, simulator: Simulator, input_dir: Path):
+        """Setup the scenario configuration files and arguments."""
+        self.gen_aux_files(simulator, input_dir)
+        self.gen_config(simulator, input_dir)
+
+        args = ()
+        config_filename = self.get_config_filename(simulator)
+        if config_filename:
+            args += (config_filename,)
+        return args
+
+    @abstractmethod
+    def gen_aux_files(self, simulator: Simulator, input_dir: Path):
+        """Generate the auxiliary files for the scenario. To be implemented
+        in subclasses."""
+        pass
+
+    @abstractmethod
+    def gen_config(self, simulator: Simulator, input_dir: Path):
+        """Generate the configuration files for the scenario. To be implemented
+        in subclasses."""
+        pass
```

Comment:
The docstrings here should be one-liners. Maybe remove the bit "To be implemented in subclasses."?